grails-webhooks/grails-app/controllers/webhooks/WebhookController.groovy

```diff
@@ -175,4 +182,24 @@ class WebhookController {
         if(action != AuthConstants.ACTION_ADMIN) authorizedActions.add(action)
         rundeckAuthContextEvaluator.authorizeProjectResourceAny(authContext,AuthConstants.RESOURCE_TYPE_WEBHOOK,authorizedActions,project)
     }
+
+    @PackageScope
+    boolean authorizeWebhookSecret(AuthContext authContext, Webhook hook, String headerAuthValue) {
+        if(!hook.secret || hook.secret.isEmpty()) return true
+        if(hook.secret.startsWith("keys/")) {
+            try {
+                def keyStorageService = storageService.storageTreeWithContext(authContext)
+                ResourceMeta contents = keyStorageService.getResource(hook.secret).getContents();
+                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                contents.writeContent(byteArrayOutputStream);
+
+                return headerAuthValue == new String(byteArrayOutputStream.toByteArray());
+            } catch (IOException e) {
```

Comment:
I think this also needs to catch `StorageException` to handle auth failure and not found